core/src/main/java/org/elasticsearch/common/rounding/TimeZoneRounding.java

```diff
@@ -218,7 +219,19 @@ public byte id() {
         public long roundKey(long utcMillis) {
             long timeLocal = timeZone.convertUTCToLocal(utcMillis);
             long rounded = Rounding.Interval.roundValue(Rounding.Interval.roundKey(timeLocal, interval), interval);
-            return timeZone.convertLocalToUTC(rounded, false, utcMillis);
+            try {
+                return timeZone.convertLocalToUTC(rounded, true, utcMillis);
+            } catch (IllegalInstantException e) {
```

Comment:
Note: I don't really like to detect this special case by exception, but the only other way I see how this can be detected is copy the underlying Joda code here or try to use a heuristic like (timeZone.convertLocalToUTC(rounded) > utcMillis) to detect that the rounded value falls into the DST gap.